src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticsPublisher.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    internal class RazorDiagnosticsPublisher : DocumentProcessedListener
```

Comment:
I feel like this is too specific. I'd imagine we might do other things in the future that listen to document processed events. I suggest having a `DefaultDocumentProcessedListener` and having that call into `RazorDiagnosticPublisher`.

Reply:
Not quite sure I follow the suggestion. Could you clarify a bit more?

Reply:
Imagine we want to implement another feature for which we need to know when the document processing is complete. How would you hook that up? Since `DocumentProcessedListener` is a singleton, we would already have an implementation of it (`RazorDiagnosticPublisher`) registered.

Reply:
Ahh, you missed this piece: https://github.com/aspnet/Razor.VSCode/pull/277/files#diff-71b7bdf65ccf5bc35df26932bdc6b4dfR31

You can have multiple `DocumentProcessedListener`'s, it imports an `IEnumerable` of them 😄

Reply:
Ah true. This is fine then 😄